cms/djangoapps/contentstore/views/item.py

```diff
@@ -1250,3 +1258,54 @@ def _xblock_type_and_display_name(xblock):
     return _('{section_or_subsection} "{display_name}"').format(
         section_or_subsection=xblock_type_display_name(xblock),
         display_name=xblock.display_name_with_default)
+
+
+def xblock_summary(xblock, include_children=False):
+    """
+    Returns a summarized dict of an xblock.
+
+    Arguments:
+        xblock (Xblock): An XBlock whose summary is to be made.
+        include_children (bool): If True, includes child XBlocks information.
+
+    Returns:
+        xblock_info(dict): A dict containing basic information about the XBlock.
+
+        A sample output may look like :
+        {
+            'category': 'sequential',
+            'display_name': 'seq1',
+            'location': 'i4x://org.0/course_0/sequential/5e828fac0c0d40879efeda6e3b6d1db5',
+            'child_info': [
+                {
+                    'category': 'vertical',
+                    'display_name': 'vertical1',
+                    'location': 'i4x://org.0/course_0/vertical/81ded1bb165e4b6fa19623911869a89d',
+                    'child_info': [
+                        {
+                            'category': 'problem',
+                            'display_name': 'problem1',
+                            'location': 'i4x://org.0/course_0/problem/2df788b55e0848e596859a20010c9f71'
+                        }, {
+                            'category': 'html',
+                            'display_name': 'html1',
+                            'location': 'i4x://org.0/course_0/html/7d74d1987128499d96530ac3e431eb1c'
+                        }
+                    ]
+                },
+                {
+                    'category': 'vertical',
+                    'display_name': 'vertical2',
+                    'location': 'i4x://org.0/course_0/vertical/81ded1bb165e4b6fa19623911869a8d2'
+                }
+            ]
+        }
+    """
+    xblock_info = {
+        'location': unicode(xblock.location),
+        'display_name': xblock.display_name,
+        'category': xblock.category
+    }
+    if include_children and xblock.has_children:
+        xblock_info['child_info'] = [xblock_summary(child) for child in xblock.get_children()]
```

Comment:
Shouldn't this pass ``include_children`` to the ``xblock_summary`` in order to get the full tree? The default is not to include the children.